projects/npm-tools/packages/npm-scripts/src/utils/buildArtifacts.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * SPDX-FileCopyrightText: © 2022 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+
+const expandGlobs = require('./expandGlobs');
+const findRoot = require('./findRoot');
+
+const CWD = process.cwd();
+const root = findRoot();
+
+const npmScriptsCacheDir = path.join(root, '.npmscripts', 'build');
+const artifactCache = path.join(npmScriptsCacheDir, 'artifacts');
+
+const BUILD_INFO_FILE_NAME = 'buildinfo.json';
+
+const hashFile = (filePath) => {
+	const fileBuffer = fs.readFileSync(filePath);
+	const hashSum = crypto.createHash('sha256');
```

Comment:
A SHA-1 would probably be enough for these purposes and may be faster. 

However, I'm OK with SHA-256 as it won't probably take too much longer.